Examples/ArcGISToolkitExamples/JobManagerExample.swift

```diff
@@ -0,0 +1,373 @@
+        }
+        else{
+
+            let portalItem = AGSPortalItem(url: URL(string:"https://www.arcgis.com/home/item.html?id=acc027394bc84c2fb04d1ed317aac674")!)!
+            let map = AGSMap(item: portalItem)
+            // naperville
+            let env = AGSEnvelope(xMin: -9825684.031125, yMin: 5102237.935062, xMax: -9798254.961608, yMax: 5151000.725314, spatialReference: AGSSpatialReference.webMercator())
+            takeOffline(map: map, extent: env)
+        }
+        
+        i += 1
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    override open func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return jobs.count
+    }
+    
+    override open func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "JobCell") as! JobTableViewCell
+        let job = jobs[(indexPath as NSIndexPath).row]
+        cell.configureWithJob(job: job)
+        return cell
+    }
+    
+    open func tableView(_ tableView: UITableView, didSelectRowAtIndexPath indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+    
+    var documentsPath: String{
+        return NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0]
+    }
+    
+    func generateGDB(URL: URL, syncModel: AGSSyncModel, extent: AGSEnvelope?){
+        
+        let task = AGSGeodatabaseSyncTask(url: URL)
+        
+        // hold on to task so that it stays retained while it's loading
+        self.tasks.append(task)
+        
+        task.load{ [weak self, weak task] error in
+
+            // make sure we are still around...
+            guard let strongSelf = self else {
+                return
+            }
+            
+            guard let strongTask = task else{
+                return
+            }
+            
+            // remove task from array now that it's loaded
+            if let index = strongSelf.tasks.index(where: {return $0 === strongTask}){
+                strongSelf.tasks.remove(at: index)
+            }
+            
+            // return if error or no featureServiceInfo
+            guard error == nil else{
+                print("error: \(error!)")
+                return
+            }
+            
+            guard let fsi = strongTask.featureServiceInfo else{
+                return
+            }
+            
+            let params = AGSGenerateGeodatabaseParameters()
+            
+            params.extent = extent
+            if params.extent == nil{
+                params.extent = fsi.fullExtent
+            }
+            
+            params.outSpatialReference = AGSSpatialReference.webMercator()
```

Comment:
should this be hardcoded to read from somewhere?

Reply:
no - the point of the sample is to just kick off jobs, doesn't matter here